homeworks/roman.strokan_Strokan/homework_5/task4.js

```diff
@@ -0,0 +1,21 @@
+const people = [
+    { id: 1, name: 'Nick', friends: [2, 5, 6] },
+    { id: 2, name: 'John', friends: [1, 3] },
+    { id: 3, name: 'Mike', friends: [2, 5] },
+    { id: 4, name: 'Janny', friends: null },
+    { id: 5, name: 'Edward', friends: [1, 3] },
+    { id: 6, name: 'Jeen', friends: [5, 1] },
+];
+
+function getPeople(userId) {
+    const array = [];
+    const usedObj = people.find(currentObj => currentObj.id === userId);
+    if (userId < Object.keys(people).length && usedObj.friends !== null) {
```

Comment:
why do you even care about `length`? it could be anything, like `userId` could be anything as well

wrong approach

Reply:
> why do you even care about `length`? it could be anything, like `userId` could be anything as well
> 
> wrong approach

Я добавил эту проверку не для того чтобы ограничить. Если входящее значение превышает колличество пользователей я получу краш. У нас 6 пользователей. Если я передам в функцию 100500, скрипт ляжет.  Или я не правильно понял суть моей ошибки?

Reply:
you compare irrelevant values... in real life `id` is just an identifier, it could be anything like `df561frt67aa`... and you take comparing this to array length as a basis to your logical operation... this is totally wrong

you just need:

1. find user by provided id
2. filter original array by this user's friends ids

that's it... no unicorns ;)